src/bot/commands/random/dog.ts

```diff
@@ -0,0 +1,214 @@
+    "mexicanhairless",
+    "mix",
+    "mountain-bernese",
+    "mountain-swiss",
+    "newfoundland",
+    "otterhound",
+    "papillon",
+    "pekinese",
+    "pembroke",
+    "pinscher-miniature",
+    "pointer-german",
+    "pointer-germanlonghair",
+    "pomeranian",
+    "poodle-miniature",
+    "poodle-standard",
+    "poodle-toy",
+    "pug",
+    "puggle",
+    "pyrenees",
+    "redbone",
+    "retriever-chesapeake",
+    "retriever-curly",
+    "retriever-flatcoated",
+    "retriever-golden",
+    "ridgeback-rhodesian",
+    "rottweiler",
+    "saluki",
+    "samoyed",
+    "schipperke",
+    "schnauzer-giant",
+    "schnauzer-miniature",
+    "setter-english",
+    "setter-gordon",
+    "setter-irish",
+    "sheepdog-english",
+    "sheepdog-shetland",
+    "shiba",
+    "shihtzu",
+    "spaniel-blenheim",
+    "spaniel-brittany",
+    "spaniel-cocker",
+    "spaniel-irish",
+    "spaniel-japanese",
+    "spaniel-sussex",
+    "spaniel-welsh",
+    "springer-english",
+    "stbernard",
+    "terrier-american",
+    "terrier-australian",
+    "terrier-bedlington",
+    "terrier-border",
+    "terrier-dandie",
+    "terrier-fox",
+    "terrier-irish",
+    "terrier-kerryblue",
+    "terrier-lakeland",
+    "terrier-norfolk",
+    "terrier-norwich",
+    "terrier-patterdale",
+    "terrier-russell",
+    "terrier-scottish",
+    "terrier-sealyham",
+    "terrier-silky",
+    "terrier-tibetan",
+    "terrier-toy",
+    "terrier-westhighland",
+    "terrier-wheaten",
+    "terrier-yorkshire",
+    "vizsla",
+    "weimaraner",
+    "whippet",
+    "wolfhound-irish"
+]
+
+export class Dog extends Command {
+    constructor() {
+        super({
+            name: 'dog',
+            description: 'Displays a random image of a dog.',
+            arguments: [
+                {
+                    name: 'breed',
+                    description: 'The type of dog to get an image of.',
+                    options: breeds,
+                    error: true,
+                    default: '',
+                }
+            ]
+        });
+    }
+
+    async execute(input: Input) {
+        let breed = input.getArgument('breed') as string;
+        let url = 'https://dog.ceo/api/breed';
+        if(breed == '') {
+            url = `${url}s/image/random`;
+        }
+        else{
+            url = `${url}/${encodeURIComponent(breed)}/images/random`
+        }
+
+        console.log(url);
+        let message = await input.channel.send(`${Emoji.LOADING}  Fetching image...`) as Message;
+
+        //Fetch from API
+        request(url, async (err, response, body) => {
+            //Handle HTTP errors
+            if (err) {
+                await message.edit(`${Emoji.ERROR}  Failed to get image, try again later.`);
+                return;
+            }
+
+            //Parse the body
+            let parsed = <ApiResponse>JSON.parse(body);
+
+            // Delete the loading message
+            try { await message.delete(); } catch(err) {}
```

Comment:
Fun fact... You can eliminate the need for a try-catch by using `deleteAfter(0)`, which catches errors automagically.

```suggestion
            message.deleteAfter(0);
```